### PR TITLE
Release Lit v3 pre-release docs with blog post

### PR DESCRIPTION
**Pull request by AndrewJakubowicz**
Adds the Lit v3 pre-release documentation. Fixes https://github.com/lit/lit.dev/issues/1109

1. Copied all the v2 docs to v3.
2. Then fixed all cross links `/docs/v2` to `/docs/v3`.
3. Replaced `{% api ` with `{% api-v3` which ensures links to API go to v3 api.
4. Updated the upgrade guide page with Lit 3.0 specific guidance.
5. Added a pre-release banner.
6. Made it so that switching between v2 and v3 keeps you on the same page (using manual `versionLinks`). Note this system does not support generated API pages.

Requirements pages were updated by @augustjk in https://github.com/lit/lit.dev/pull/1118 - Thank you!

Tested manually & via existing tests.

### CODE REVIEW

**Comment by AndrewJakubowicz** [comment]
TODO: Hide from search index.

File: packages/lit-dev-content/site/docs/v3/releases/upgrade.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+## Overview
+
+Lit 3.0 has very few breaking changes from Lit 2.0. Any projects using Lit 2
+without deprecation warnings should not require any changes to upgrade to Lit 3.
+
+Lit 3.0:
+
+- Drops support for IE11
+- Is published as ES2021
+- Removes a couple deprecated APIs
```
Comment (augustjk) [inline packages/lit-dev-content/site/docs/v3/releases/upgrade.md:20]:
```suggestion
- Removes a few deprecated APIs
```

File: packages/lit-dev-content/site/docs/v3/releases/upgrade.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+## List of removed APIs
+
+If your Lit 2.x project does not have deprecation warnings you should not be
+impacted by this list.
+
+ - [Removed `UpdatingElement` alias for `ReactiveElement`.](#removed-updating-element)
+ - [Removed re-export of decorators from main `lit-element` module.](#removed-re-export-decorators)
+ - [Removed deprecated call signature for the `queryAssignedNodes` decorator.](#removed-queryassignednodes-non-object)
+ - [Moved experimental server side rendering hydrate modules from `lit`, `lit-element`, and `lit-html` to `@lit-labs/ssr-client`.](#moved-experimental-hydration)
```
Comment (augustjk) [inline packages/lit-dev-content/site/docs/v3/releases/upgrade.md:41]:
```suggestion
 - [Moved experimental server side rendering hydration modules from `lit`, `lit-element`, and `lit-html` to `@lit-labs/ssr-client`.](#moved-experimental-hydration)
```

**Review by augustjk — APPROVED** [review]
A couple minor suggestions

**Pull request merged**